code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -617,6 +619,21 @@ var/list/ai_verbs_default = list(
 			user.visible_message("<span class='notice'>\The [user] finishes fastening down \the [src]!</span>")
 			anchored = 1
 			return
+	else if(isCrowbar(W) && user.a_intent != I_HURT)
+		if(!length(stock_parts))
+			to_chat(user, SPAN_WARNING("No parts left to remove"))
+			return
+		var/obj/item/stock_parts/remove = input(user, "Which component do you want to pry out?", "Remove Component") as null|anything in stock_parts
+		if(!remove || !(remove in stock_parts))
+			return
```

Comment:
Need to re-check CanPhysicallyInteract or similar.

Reply:
that's topic proc, I guess I can check adjacency

Reply:
Added adjcancy check